cfme/tests/cloud/test_cloud_proxy.py

```diff
@@ -0,0 +1,175 @@
+    with error.expected('Credential validation was not successful:'):
+        validate_provider(appliance, provider)
+        flash.assert_message_match('Credential validation was successful')
+
+
+def test_proxy_remove(appliance, provider, request):
+    """Removes the specific valid proxy for all providers
+
+        Args:
+            None - Uses only the build in objects
+    """
+    logger.info("Begin Test to Remove Proxy")
+
+    @request.addfinalizer
+    def _cleanup_proxy():
+        reset_proxy(appliance, [provider.type], provider)
+
+    appliance.ssh_client.run_command(
+        "rm -f /var/www/miq/vmdb/config/settings.local.yml")
+    appliance.restart_evm_service(wait_for_web_ui=True)
+    validate_provider(appliance, provider)
+    flash.assert_message_match('Credential validation was successful')
+
+
+def test_proxy_override(appliance, provider, request):
+    """ Configures a bad default and then a valid proxy for each provider
+        This one is a little weird because all the other tests setup a specific provider, but
+        this one needs to setup the default incorrectly and the provider correctly so that we know
+        the provider proxy was definitely being used and not the default.
+
+        Args:
+            None - Uses only the build in objects
+    """
+    logger.info("Begin Test Override Proxy for Provider {}".format(provider.type))
+
+    @request.addfinalizer
+    def _cleanup_proxy():
+        reset_proxy(appliance, ['default', provider.type], provider)
+
+    logger.info("Test to fail first")
+    settings_local(appliance, ["default"], False)
+    appliance.restart_evm_service(wait_for_web_ui=True)
+    with error.expected('Credential validation was not successful:'):
+        validate_provider(appliance, provider)
+        flash.assert_message_match('Credential validation was successful')
+
+    logger.info("Then test to pass using provider specific override")
+    settings_local(appliance, ['default', provider.type], True)
+    appliance.restart_evm_service(wait_for_web_ui=True)
+    validate_provider(appliance, provider)
+    flash.assert_message_match('Credential validation was successful')
+
+
+def settings_local(appliance, proxy_type, valid=True):
+    """Creates the file string and sends it to the appliance.  If more than one provider,
+        It loops through each item in proxy_type
+
+        Args: proxy_type - is a list that is either or more provider.type and/or 'default'
+              valid - determines if we want the proxy to pass or fail.
+
+    """
+    proxy = cfme_data['proxy_servers']['default']
+    creds = credentials[proxy['credentials']]
+    yml_header = ":http_proxy: \n  "
+    for pt in proxy_type:
+        if len(proxy_type) > 1 and pt == 'default':
+            port = '9999'
+        else:
+            port = proxy['port'] if valid else '8686'
+        yml_provider = ":{pt}: \n    :host: {host} \n    ".format(pt=pt, host=proxy['host'])
```

Comment:
We should not be constructing yaml format with string manipulation.Moreover there is a yaml module where we can pass it a dict/list and it will create it for us.

However I believe that the ```set_yaml_config``` actually takes a dict anyway to update the settings and doesn't require you to turn it into readable yaml.